lib/maze.rb

```diff
@@ -8,16 +8,25 @@
 # providing an alternative to the proliferation of global variables or singletons.
 module Maze
 
-  VERSION = '9.21.0'
+  VERSION = '9.22.0'
 
   class << self
-    attr_accessor :check, :driver, :internal_hooks, :mode, :start_time, :dynamic_retry, :public_address,
+    attr_accessor :check, :internal_hooks, :mode, :start_time, :dynamic_retry, :public_address,
                   :public_document_server_address, :run_uuid, :scenario
 
     def config
       @config ||= Maze::Configuration.new
     end
 
+    def driver
+      raise 'Cannot use a failed driver' if @driver&.failed?
```

Comment:
Will this raise cause each attempt to acquire the driver just raise an error when called, but not actually return anything for logical checks?

Reply:
So trying to call `Maze.driver.failed?` will simply raise the error and not actually return the driver, so it won't trigger checks like `if Maze.driver.failed?`.

Could we simply log instead, but allow the driver to be returned and checked?